apps/src/dance/ai/DanceAiModalHeader.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useMemo} from 'react';
+import moduleStyles from './dance-ai-modal.module.scss';
+import aiBotMini from '@cdo/static/dance/ai/bot/ai-bot-mini.svg';
+const i18n = require('../locale');
+
+import {DanceAiModelItem} from './types';
+
+import EmojiIcon from './EmojiIcon';
+
+interface DanceAiModalHeaderProps {
+  onClose: () => void;
+  handleStartOverClick: () => void;
+  selectedEmojis: DanceAiModelItem[];
+}
+
+const DanceAiModalHeader: React.FunctionComponent<DanceAiModalHeaderProps> = ({
+  onClose,
+  handleStartOverClick,
+  selectedEmojis,
+}) => {
+  const headerValue = useMemo(
+    () => (
+      <div
+        className={moduleStyles.inputsContainer}
+        tabIndex={0}
+        onClick={handleStartOverClick}
+      >
+        {selectedEmojis.map(item => (
+          <div key={item.id} className={moduleStyles.emojiSlot}>
+            {item && (
+              <EmojiIcon item={item} className={moduleStyles.emojiSlotIcon} />
+            )}
+          </div>
+        ))}
+      </div>
+    ),
+    [handleStartOverClick, selectedEmojis]
+  );
```

Comment:
This was originally a function that was only called once. That would cause e-renders, so was better as a `useCallback`. But since it was only called once, was better as a `useMemo`

Should also probably be combined with `headerContent` down below into a single `useMemo` instead.